### PR TITLE
Add demo-magic demo for multi-tenancy

### PR DESCRIPTION
**Pull request by sambhav**
Signed-off-by: GitHub <noreply@github.com>

### CODE REVIEW

File: multitenancy/demo.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+source <(curl -s -L https://raw.githubusercontent.com/paxtonhare/demo-magic/master/demo-magic.sh)
```
[the remark this answers is not in the document]
Comment (sambhav) [reply multitenancy/demo.sh:3]: @sonatype-lift ignore

File: multitenancy/demo.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+source ../demo-magic.sh
+
+{
+    cat *.yaml | kubectl delete --ignore-not-found -f - 1> /dev/null
```
[the remark this answers is not in the document]
Comment (sambhav) [reply multitenancy/demo.sh]: @sonatype-lift ignore

File: multitenancy/demo.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+source ../demo-magic.sh
```
[the remark this answers is not in the document]
Comment (sambhav) [reply multitenancy/demo.sh]: @sonatype-lift ignore

File: image_verification/demo.sh [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+source <(curl -s -L https://raw.githubusercontent.com/paxtonhare/demo-magic/master/demo-magic.sh)
+
+{
+    cat *.yaml | kubectl delete --ignore-not-found -f -
+    kubectl delete ns demo --ignore-not-found
+    kubectl create ns demo
+} 1> /dev/null 2>&1
+
+TYPE_SPEED=35
```
[the remark this answers is not in the document]
Comment (sambhav) [reply image_verification/demo.sh:11]: @sonatype-lift ignore

File: image_verification/demo.sh [hunk 5]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+source <(curl -s -L https://raw.githubusercontent.com/paxtonhare/demo-magic/master/demo-magic.sh)
+
+{
+    cat *.yaml | kubectl delete --ignore-not-found -f -
```
[the remark this answers is not in the document]
Comment (sambhav) [reply image_verification/demo.sh:6]: @sonatype-lift ignore

File: image_verification/demo.sh [hunk 6]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+source <(curl -s -L https://raw.githubusercontent.com/paxtonhare/demo-magic/master/demo-magic.sh)
```
[the remark this answers is not in the document]
Comment (sambhav) [reply image_verification/demo.sh:3]: @sonatype-lift ignore

File: multitenancy/demo.sh [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+source <(curl -s -L https://raw.githubusercontent.com/paxtonhare/demo-magic/master/demo-magic.sh)
+
+{
+    cat *.yaml | kubectl delete --ignore-not-found -f -
+    kubectl delete ns test --ignore-not-found
+} 1> /dev/null 2>&1
+
+
+pe "cat *.yaml | kubectl apply -f -"
```
Comment (JimBugwadia) [inline multitenancy/demo.sh:13]: I forgot to mention, but to add the policy which creates cluster-wide roles Kyverno needs additional permissions. I typically add the permissions as part of the demo, as its a good item to showcase.

File: image_verification/check_attestations.yaml [hunk 8]
```diff
@@ -42,10 +43,7 @@ spec:
               - key: "{{ scanner }}"
                 operator: Equals
                 value: trivy
-              - key: "{{ time_since('', timestamp, '') }}"
-                operator: LessThan
-                value: 24h
-              - key: "{{ Results[].Vulnerabilities[?CVSS.redhat.V3Score > `8.0`][] | length(@) }}"
+              - key: "{{ Results[].Vulnerabilities[?CVSS.redhat.V3Score >= `10.0`][] | length(@) }}"
```
Comment (JimBugwadia) [inline image_verification/check_attestations.yaml:49]: For the demo, its nice to typically show that a resource with an older scan is blocked and allow a resource with an up to date scan to be allowed.

Same for the CVSS score.

Should we have two policies, or maybe have a way to patch the existing policy via kubectl?

**Review by JimBugwadia — CHANGES_REQUESTED** [review]
couple of questions...

File: multitenancy/demo.sh (hunk 7 above)
Comment (sambhav) [reply multitenancy/demo.sh:13]: Is it okay if we skip this for the presentation on Wednesday given that the timing is tight. I have set up the cluster to have appropriate permissions for now, but I can mention it during the demo that the generate controller will need appropriate permissions and that can be customized through the helm chart.

File: image_verification/check_attestations.yaml (hunk 8 above)
Comment (sambhav) [reply image_verification/check_attestations.yaml:49]: Added a test with 2 diff. images.

File: multitenancy/demo.sh (hunk 7 above)
Comment (sambhav) [reply multitenancy/demo.sh:13]: Added an extra RBAC file.

**Review by JimBugwadia — APPROVED** [review]

**Pull request merged**